src/sqlfluff/dialects/dialect_snowflake.py

```diff
@@ -958,7 +958,53 @@ class AlterTableTableColumnActionSegment(BaseSegment):
 
     match_grammar = Sequence(
         OneOf(
-            # @TODO: Add Column
+            # Add Column
+            Sequence(
+                "ADD",
+                "COLUMN",
+                Ref("ColumnReferenceSegment"),
+                Ref("DatatypeSegment"),
+                OneOf(
+                    # Default
+                    Sequence(
+                        "DEFAULT",
+                        Ref("ExpressionSegment"),
+                    ),
+                    # Auto-increment/identity column
+                    Sequence(
+                        OneOf(
+                            Ref.keyword("AUTOINCREMENT"),
+                            Ref.keyword("IDENTITY"),
```

Comment:
Do you need these to be `Ref`s?

```suggestion
                            "AUTOINCREMENT", "IDENTITY",
```

Reply:
For my context: When should we use `Ref.keyword(<keyword>)` versus just `<keyword>`?

Reply:
`Ref.keyword(<keyword>)` is needed when you need extra params (primarily `optional=True`): `Ref.keyword(<keyword>, optional=True)`